test/integration/fake_upstream.h

```diff
@@ -436,7 +442,7 @@ class FakeHttpConnection : public Http::ServerConnectionCallbacks, public FakeCo
     Network::FilterStatus onData(Buffer::Instance& data, bool) override {
       Http::Status status = parent_.codec_->dispatch(data);
 
-      if (Http::isCodecProtocolError(status)) {
+      if (Http::isCodecProtocolError(status) && !parent_.expect_protocol_error_) {
```

Comment:
I see this is the fix? I would think it would work though with the flush and write delay?

Reply:
I answered in another comment too, but the issue isn't the flushing but the connection close prematurely causing my integration test to advance past `waitForDisconnect()`. I want to wait until the client closes rather than the upstream